### PR TITLE
[release/v1.0] Pin EnvoyProxy and RateLimit images

### PR DESCRIPTION
**Pull request by Xunzhuo**
+ Pin "envoyproxy/envoy:distroless-v1.29-latest"
+ Pin "envoyproxy/ratelimit:19f2079f"

### CODE REVIEW

**Review by zirain — APPROVED** [review]

File: api/v1alpha1/shared_types.go [hunk 1]
```diff
@@ -20,15 +20,15 @@ const (
 	// DefaultDeploymentMemoryResourceRequests for deployment memory resource
 	DefaultDeploymentMemoryResourceRequests = "512Mi"
 	// DefaultEnvoyProxyImage is the default image used by envoyproxy
-	DefaultEnvoyProxyImage = "envoyproxy/envoy:distroless-dev"
+	DefaultEnvoyProxyImage = "envoyproxy/envoy:distroless-v1.29-latest"
```
Comment (zhaohuabing) [inline api/v1alpha1/shared_types.go:23]: Will distroless-v1.29-latest changes when 1.29.2 comes out? Should we use distroless-v1.29.1 or distroless-v1.29-latest ?

Comment (Xunzhuo) [reply api/v1alpha1/shared_types.go:23]: This logic is same as how we pin envoyproxy in v0.6 release.

Comment (Xunzhuo) [reply api/v1alpha1/shared_types.go:23]: https://github.com/envoyproxy/gateway/pull/2098

**Pull request merged**